### PR TITLE
fix Wunused-variable warning

### PR DESCRIPTION
**Pull request by chaychen2005**

### CODE REVIEW

File: libblockchain/BlockChainInterface.h [hunk 1]
```diff
@@ -73,8 +73,7 @@ class BlockChainInterface
         dev::eth::Block& block, std::shared_ptr<dev::blockverifier::ExecutiveContext>) = 0;
     virtual std::pair<int64_t, int64_t> totalTransactionCount() = 0;
     virtual dev::bytes getCode(dev::Address _address) = 0;
-    virtual void getNonces(std::vector<dev::eth::NonceKeyType>& _nonceVector, int64_t _blockNumber)
-    {}
+    virtual void getNonces(std::vector<dev::eth::NonceKeyType>&, int64_t) {}
```
Comment (bxq2011hust) [inline libblockchain/BlockChainInterface.h]: What about `virtual void getNonces(std::vector<dev::eth::NonceKeyType>&, int64_t) = 0`

File: libsync/SyncInterface.h [hunk 2]
```diff
@@ -54,9 +54,7 @@ class SyncInterface
     virtual void setProtocolId(PROTOCOL_ID const _protocolId) = 0;
 
     // verify handler to check downloading block
-    virtual void registerConsensusVerifyHandler(
-        std::function<bool(dev::eth::Block const&)> _handler)
-    {}
+    virtual void registerConsensusVerifyHandler(std::function<bool(dev::eth::Block const&)>) {}
```
Comment (bxq2011hust) [inline libsync/SyncInterface.h]: what about `virtual void registerConsensusVerifyHandler(std::function<bool(dev::eth::Block const&)>)=0`

File: libblockchain/BlockChainInterface.h (hunk 1 above)
Comment (chaychen2005) [reply libblockchain/BlockChainInterface.h]: has used pure virtual function

File: libsync/SyncInterface.h (hunk 2 above)
Comment (chaychen2005) [reply libsync/SyncInterface.h]: has used pure virtual function

**Pull request merged**